### PR TITLE
[curl] added completion spec skeleton

### PR DESCRIPTION
**Pull request by Nutlope**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
feature
**What is the current behavior? (You can also link to an open issue here)**
only supports 3 options
**What is the new behavior (if this is a feature change)?**
now supports all options
**Additional info:**

### CODE REVIEW

File: dev/curl.ts [hunk 1]
```diff
@@ -15,19 +181,705 @@ export const completionSpec: Fig.Spec = {
...
+    },
+    {
+      name: "--abstract-unix-socket",
+      description: "Connect via abstract Unix domain socket",
+      args: { name: "path" },
+    },
+    {
+      name: "--alt-svc",
+      description: "name> Enable alt-svc with this cache file",
+      args: { name: "fil" },
```
Comment (Monkeyank) [inline dev/curl.ts:245]: Should this be template of file paths, even if it isn't change to `name: "file"`

File: dev/curl.ts [hunk 2]
```diff
@@ -15,19 +181,705 @@ export const completionSpec: Fig.Spec = {
...
+      description: "Certificate file type",
+      args: {
+        name: "type",
+        suggestions: [{ name: "DER" }, { name: "PEM" }, { name: "ENG" }],
+      },
+    },
+    {
+      name: "--ciphers",
+      description: "of ciphers> SSL ciphers to use",
+      args: { name: "lis" },
```
Comment (Monkeyank) [inline dev/curl.ts:274]: `"list`?

File: dev/curl.ts [hunk 3]
```diff
@@ -15,19 +181,705 @@ export const completionSpec: Fig.Spec = {
...
+    { name: "--proxy-tlsv1", description: "Use TLSv1 for HTTPS proxy" },
+    {
+      name: "--proxy1.0",
+      description: "Use HTTP/1.0 proxy on given port",
+      args: { name: "host[:port]" },
+    },
+    {
+      name: "--pubkey",
+      description: "SSH Public key file name",
+      args: { name: "key" },
```
Comment (Monkeyank) [inline dev/curl.ts:722]: Should this be looking for filepaths?

File: dev/curl.ts [hunk 4]
```diff
@@ -15,19 +181,705 @@ export const completionSpec: Fig.Spec = {
...
+    { name: "--tftp-no-options", description: "Do not send any TFTP options" },
+    {
+      name: "--tls-max",
+      description: "Set maximum allowed TLS version",
+      args: { name: "VERSION" },
+    },
+    {
+      name: "--tls13-ciphers",
+      description: "of TLS 1.3 ciphersuites> TLS 1.3 cipher suites to use",
+      args: { name: "lis" },
```
Comment (Monkeyank) [inline dev/curl.ts:843]: `list`?

File: dev/curl.ts [hunk 5]
```diff
@@ -15,19 +181,705 @@ export const completionSpec: Fig.Spec = {
...
+    },
+    {
+      name: "--doh-url",
+      description: "Resolve host names over DOH",
+      args: { name: "URL" },
+    },
+    {
+      name: "--egd-file",
+      description: "EGD socket path for random data",
+      args: { name: "file" },
```
Comment (Monkeyank) [inline dev/curl.ts:358]: template of `file paths`?

**Review by Monkeyank — CHANGES_REQUESTED** [review]

**Comment by Nutlope** [comment]
Thanks for the review @Monkeyank!

**Pull request merged**